### PR TITLE
Integrate to the Inari contract

### PR DESCRIPTION
**Pull request by msilucifer**
Integration to the Inari contract
For this issue (https://github.com/sushiswap/sushiswap-interface/issues/64)

### CODE REVIEW

File: src/hooks/useInari.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+const { BigNumber } = ethers
+
+const useInari = () => {
+    const { account } = useActiveWeb3React()
+
+    const addTransaction = useTransactionAdder()
+    const sushiContract = useSushiContract(true) // withSigner
+    const inariContract = useInariContract(true)
+    const bentoBoxContract = useBentoBoxContract(true)
+    const aaveContract = useAaveContract(true)
```
[the remark this answers is not in the document]
Comment (omakasebar) [reply src/hooks/useInari.ts:18]: Cool. Seems a bit heavy to include the ABI for a contract if its just an address string. How about we just make a CONST AaaveAddress = "<checksummed address>" and remove the contract abis and hooks. Thoughts?

[the remark this answers is not in the document]
Comment (msilucifer) [reply src/hooks/useInari.ts]: There were some misunderstanding things about the `Inari` contract but it's all been fixed now.

**Comment by omakasebar** [comment]
Please fetch and merge this branch with latest master

A few issues, I encountered while testing:
1. Unable to withdraw/unwind axSUSHI - clicking withdraw produces an execution revert
axWithdraw

2. crXSushi Balance reads 0, even though I have some in my wallet. Balance is probably fetched from wrong address
crXSUSHI

3. Bento Balances are all not displayed

4. Approve tx messages could be more informative: Approve Inari for example
Screen Shot 2021-05-07 at 1 53 21 AM

**Comment by msilucifer** [comment]
```crXSushi Balance reads 0, even though I have some in my wallet. Balance is probably fetched from wrong address```
No, I set the correct address as crXSushi token address (0x228619cca194fbe3ebeb2f835ec1ea5080dafbb2).

**Comment by omakasebar** [comment]
In Inari does crXSushi become staked (out of the wallet?). If this is the case, then you will need another method to determine a user's staked crXsushi

**Comment by omakasebar** [comment]
> `crXSushi Balance reads 0, even though I have some in my wallet. Balance is probably fetched from wrong address`
> No, I set the correct address as crXSushi token address (0x228619cca194fbe3ebeb2f835ec1ea5080dafbb2).
In Inari does crXSushi become staked (out of the wallet?). If this is the case, then you will need another method to determine a user's staked crXsushi

**Comment by Ro5s** [comment]
> > `crXSushi Balance reads 0, even though I have some in my wallet. Balance is probably fetched from wrong address`
> > No, I set the correct address as crXSushi token address (0x228619cca194fbe3ebeb2f835ec1ea5080dafbb2).
> 
> In Inari does crXSushi become staked (out of the wallet?). If this is the case, then you will need another method to determine a user's staked crXsushi

No, crXUSHI should be returned to user wallet after calls like `stakeSushiToCream`. In calls like `stakeSushiToBento`, though, crXSUSHI is staked and the user will have a balance in BentoBox shares.

**Pull request closed without merging**